gfx/gl/GLContextProviderEGL.cpp

```diff
@@ -416,7 +416,13 @@ GLContextEGL::IsCurrent() {
 bool
 GLContextEGL::RenewSurface() {
     if (!mOwnsContext) {
+#ifdef MOZ_WIDGET_QT
```

Comment:
As mOwnsContext will be false only when environment variable "MOZ_USE_EXTERNAL_WINDOW" is set. Thus, current MOZ_LAYERS_PREFER_OFFSCREEN code path would run into the MOZ_CRASH("unimplemented") at line 428.

Execute lines 420 - 422 only if "MOZ_USE_EXTERNAL_WINDOW". Could be good time to change the environment variable to a preference, similar to "gfx.compositor.clear-context" (sha1 52a12a6e). Maybe something like:

```
DECL_GFX_PREF(Once, "gfx.compositor.external-window",          UseExternalWindow, bool, false);
```